tests/v3_api/test_gke_cluster_provisioning.py

```diff
@@ -0,0 +1,129 @@
+from common import *   # NOQA
+
+
+CREDENTIALS = os.environ.get('GKE_CREDENTIAL', "")
+RANCHER_CLEANUP_CLUSTER = os.environ.get('RANCHER_CLEANUP_CLUSTER', "True")
+GKE_MASTER_VERSION = os.environ.get('GKE_MASTER_VERSION', "1.9.7-gke.1")
+
+
+def test_create_gke_cluster():
+
+    client = get_admin_client()
+    gkeConfig = get_gke_config()
+
+    print "Cluster creation"
+    cluster = client.create_cluster(gkeConfig)
+    clusterid = cluster.id
+    print cluster
+    # Sleep added for the cluster to show the correct state
+    time.sleep(30)
+    print "Cluster list"
+    clusterlist = client.list_cluster(clusterId=clusterid)
+    cluster = client.reload(clusterlist[0])
+    print cluster.state
+    cluster = validate_cluster(client, cluster, check_intermediate_state=True,
+                               skipIngresscheck=True)
+
+    if RANCHER_CLEANUP_CLUSTER == "True":
+        delete_cluster(client, cluster)
+
+
+def test_create_gke_cluster_with_validate_endpoint_ingress():
+
+    client = get_admin_client()
+    gkeConfig = get_gke_config()
+
+    print "Cluster creation"
+    cluster = client.create_cluster(gkeConfig)
+    clusterid = cluster.id
+    print cluster
+    # Sleep added for the cluster to show the correct state
+    time.sleep(30)
+    print "Cluster list"
+    clusterlist = client.list_cluster(clusterId=clusterid)
+    cluster = client.reload(clusterlist[0])
+    print cluster.state
+
+    cluster = validate_cluster(client, cluster, check_intermediate_state=True,
+                               skipIngresscheck=True)
```

Comment:
@soumyalj Would the regular ingress creation with host set not work in this case ? Why do we need to test xip.io in this case ?

Reply:
Another test will be added for regular ingress creation. It would not harm to have this test.